scipy/sparse/linalg/_eigen/_svds.py

```diff
@@ -102,7 +102,7 @@ def _iv(A, k, ncv, tol, which, v0, maxiter,
 
         shape = (A.shape[0],) if solver == 'propack' else (min(A.shape),)
         if v0.shape != shape:
-            message = "`v0` must have shape {shape}."
+            message = f"`v0` must have shape {shape}."
```

Comment:
@mckib2 It looks like this is intended to be an f-string, right?

Reply:
That's right. I'll merge when CI finishes.

Reply:
Indeed, tests test only the beginning:
https://github.com/scipy/scipy/blob/278fcfb9700fe2e9f466fa9a1cd2c76b34feb395/scipy/sparse/linalg/_eigen/tests/test_svds.py#L169
https://github.com/scipy/scipy/blob/278fcfb9700fe2e9f466fa9a1cd2c76b34feb395/scipy/sparse/linalg/_eigen/tests/test_svds.py#L180

Reply:
Let's add the correct shape (explicitly) to the reference string of the test.

Reply:
+1 to add the correct shape to the reference string.

Here's what is can be changed to in line 169: (move it after `required_length` is defined)
```
        message = f"`v0` must have shape \\({required_length},\\)."
```

and line 180:
```
        message = f"`v0` must have shape \\({A[0]},\\)"
```

Reply:
@swallan I would prefer you add it, either in a new PR or even by modifying this PR if you have the proper rights.

Reply:
@DimitriPapadopoulos I submitted https://github.com/DimitriPapadopoulos/scipy/pull/1 to your fork with these changes, if you approve it, those commits with those changes will appear here.

Reply:
Done

Reply:
@swallan see tweak (`re.escape(message)` for readability).

Reply:
Thanks -- wasn't aware of that.

Reply:
I'm lost. How would '{shape}' be filled with `r"..."` instead of `f"..."`?

Reply:
You can combine both `fr"..."`

Reply:
Ah, didn't know that. Thanks!